cg/meta/upload/fohm/fohm.py

```diff
@@ -277,17 +317,66 @@ def sync_files_sftp(self) -> None:
             self.daily_rawdata_path.rmdir()
 
     def update_upload_started_at(self, case_id: str) -> None:
-        """Update timestamp for cases which started being processed as batch"""
+        """Update timestamp for cases which started being processed as batch."""
         if self._dry_run:
             return
-        case_obj: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
-        case_obj.analyses[0].upload_started_at = dt.datetime.now()
+        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        case.analyses[0].upload_started_at = dt.datetime.now()
         self.status_db.session.commit()
 
     def update_uploaded_at(self, case_id: str) -> None:
-        """Update timestamp for cases which uploaded successfully"""
+        """Update timestamp for a case which uploaded successfully."""
         if self._dry_run:
             return
-        case_obj: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
-        case_obj.analyses[0].uploaded_at = dt.datetime.now()
+        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        case.analyses[0].uploaded_at = dt.datetime.now()
         self.status_db.session.commit()
+
+    def parse_and_write_complementary_report(self) -> list[FohmComplementaryReport]:
+        """Parse and write a complementary report."""
+        complementary_reports_raw: list[dict] = self.get_reports_contents(self.daily_reports_list)
+        unique_complementary_reports_raw: list[dict] = remove_duplicate_dicts(
```

Comment:
I think this could be removed? The whole argument is that if we would pass the same case twice through the cli we would have it twice here as well. We could either create a callback on the cli function which validates that the same case name is not passed twice instead. That way we would warn the user as well that they have probably made a mistake with passing the same case twice.

Reply:
Sounds doable! But let's do it in another PR as I am already changing a lot in this PR. And this function is also used in the future GISAID PR.